### PR TITLE
Issue 709: Add Slow Bookkeeper Servers to Placement Policy for read ordering

### PR DESCRIPTION
**Pull request by philipsu522**
Descriptions of the changes in this PR:

Maintain a list of slow bookkeeper hosts, which are tried only after readonly bookkeeper hosts are tried. Bookkeeper hosts can be categorized as "slow" when a speculative timeout occurs on that bookkeeper host.

Master Issue:: 709

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/client/RackawareEnsemblePlacementPolicyImpl.java [hunk 1]
```diff
@@ -836,15 +922,54 @@ protected BookieNode selectRandomFromRack(String netPath, Set<Node> excludeBooki
...
         for (int i = 0; i < writeSet.size(); i++) {
-            writeSet.set(i, (writeSet.get(i) & ~MASK_BITS) % ensembleSize);
+            if ((writeSet.get(i) & MASK_BITS) == REMOTE_MASK) {
+                firstRemote = i;
+                break;
+            }
+        }
+        if (firstRemote != -1) {
+            int i = 0;
+            for (;i < remoteNodeInReorderSequence
```
Comment (yzang) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/client/RackawareEnsemblePlacementPolicyImpl.java]: We can simplify here? for(i=0; i<...; i++)

**Comment by sijie** [comment]
thank you @philipsu522 for updating the pull requests.

**Comment by sijie** [comment]
waiting for the travis ci to complete the verification.

@eolivelli can you take a look at again? I think now the rebase is correct.

**Comment by philipsu522** [comment]
Thanks a lot everyone, sorry for all the trouble in rebasing and merging

**Comment by sijie** [comment]
retest this please.

**Comment by sijie** [comment]
ci passed on jdk9, some failures on jdk8 but unrelated to this change. merging it now.

**Comment by sijie** [comment]
merged this change to master. @philipsu522 great contribution! well done.

**Pull request closed without merging**